src/components/Tag/Tag.scss

```diff
@@ -52,6 +48,20 @@ $icon-size: rem(16px);
     }
   }
 
+  &.removable {
+    padding-right: 0;
+  }
+
+  &.segmented {
+    > * {
+      max-width: 100%;
+
+      > * {
+        max-width: 97%;
```

Comment:
This is so that the content leaves some room for remove button when the it requires ellipsis... maybe this can be done in a better way?